contracts/VestAMM/libraries/AmmIntegration/CurveVestAMM.sol

```diff
@@ -173,4 +153,108 @@ contract CurveVestAMM {
 
         return curvePool.get_dy(0, 1, 1 ether);
     }
+
+    ////////////////////////////////////
+    // Virtual Price Estimation Logic //
+    ////////////////////////////////////
+
+    /**
+     * @dev A snapshot of a liquidity pool at a given time - perhaps integrate into Struct used above?
+     */
+    struct Pooldata {
+        uint256 amountA; // Raw amounts of each token in the pool - assuming only two tokens in the pool
+        uint256 amountB;
+        uint256 ratioA; // (amountA * 1e18) / amountB
+        uint256 ratioB; // (amountB * 1e18) / amountA
+        uint256 lpTotal; // Total supply of LP Tokens
+        uint256 lpAmount; // Amount of LP tokens owned
+    }
+
+    Pooldata public firstPoolData;
+    Pooldata public secondPoolData;
+
+    function getPoolData(
+        IERC20 _tokenA,
+        IERC20 _tokenB,
+        IERC20 _lpContract,
+        IERC20 _lpToken
+    ) internal returns (Pooldata memory) {
+        Pooldata memory newPoolData;
+
+        // NOTE Pass the zero address to token A in order to specify ETH instead of an ERC20
+        // Might not be the best solution to this problem, not sure
+        if (address(_tokenA) == address(0)) {
+            newPoolData.amountA = address(_lpContract).balance;
+        } else {
+            newPoolData.amountA = _tokenA.balanceOf(address(_lpContract));
+        }
+
+        newPoolData.amountB = _tokenB.balanceOf(address(_lpContract));
+
+        newPoolData.ratioA = (newPoolData.amountA * 1e18) / newPoolData.amountB;
+        newPoolData.ratioB = (newPoolData.amountB * 1e18) / newPoolData.amountA;
+
+        newPoolData.lpTotal = _lpToken.totalSupply();
+        newPoolData.lpAmount = _lpToken.balanceOf(address(this));
+
+        return newPoolData;
+    }
+
+    /**
+     * @dev Returns an estimate of lpVirtualPriceAtTime1 - lpVirtualPriceAtTime0
+     */
+    function getEstimatedVirtualPriceDelta(
+        IERC20 _tokenA,
+        IERC20 _tokenB,
+        IERC20 _lpContract,
+        IERC20 _lpToken
+    ) internal returns (uint256) {

```

Comment:
So the arguments should look like:
`getEstimatedVirtualPriceDelta(IVestAMM.VAmmInfo calldata _vammInfo,  Pooldata calldata _firstPoolData)`